plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -810,6 +795,64 @@ private function months_filter() {
 		echo '</select>';
 	}
 
+	/**
+	 * Get order year-months cache. We cache the results in the options table, since these results will change very infrequently.
+	 * We use the heuristic to always return current year-month when getting from cache to prevent an additional query.
+	 *
+	 * @return array List of year-months.
+	 */
+	private function get_and_maybe_update_months_filter_cache(): array {
+		global $wpdb;
+
+		// We cache in the options table since it's won't be invalidated soon.
+		$cache_option_value_name = 'wc_' . $this->order_type . '_list_table_months_filter_cache_value';
+		$cache_option_date_name  = 'wc_' . $this->order_type . '_list_table_months_filter_cache_date';
+
+		$cached_timestamp = get_option( $cache_option_date_name, 0 );
+
+		// new day, new cache.
+		if ( 0 === $cached_timestamp || gmdate( 'j', time() ) !== gmdate( 'j', $cached_timestamp ) || ( time() - $cached_timestamp ) > 60 * 60 * 24 ) {
+			delete_option( $cache_option_date_name );
+			delete_option( $cache_option_value_name );
```

Comment:
I wonder if we could rearrange the order of things a bit to make these two extra database operations unnecessary (since the same rows get updated further down)...